crates/matrix-sdk-ui/src/timeline/mod.rs

```diff
@@ -870,6 +871,30 @@ impl Timeline {
             Ok(false)
         }
     }
+
+    pub async fn pin_event(&self, event_id: &EventId) -> Result<bool> {
+        let mut pinned_events = self.room().pinned_events().get();
+        let event_id = event_id.to_owned();
+        if pinned_events.contains(&event_id) {
+            Ok(false)
+        } else {
+            pinned_events.push(event_id);
+            let content = RoomPinnedEventsEventContent::new(pinned_events);
+            self.room().send_state_event(content).await.map(|_| true)
```

Comment:
I would prefer `?; Ok(true)` instead of `.map(|_| true)` but happy to leave it if you prefer it the way it is.